source/core_mqtt.c

```diff
@@ -3704,6 +3723,98 @@ static MQTTStatus_t handleSubUnsubAck( MQTTContext_t * pContext,
 
 /*-----------------------------------------------------------*/
 
+static MQTTStatus_t sendDisconnectWithoutCopy( MQTTContext_t * pContext,
+                                               MQTTSuccessFailReasonCode_t * pReasonCode,
+                                               size_t remainingLength,
+                                               const MQTTPropBuilder_t * pPropertyBuilder )
+{
+    int32_t bytesSentOrError;
+    size_t ioVectorLength = 0U;
+    size_t totalMessageLength = 0U;
+    size_t disconnectPropLen = 0U;
+    MQTTStatus_t status = MQTTSuccess;
+
+    /* Maximum number of bytes required by the fixed size part of the CONNECT
+     * packet header according to the MQTT specification.
+     * MQTT Control Byte      0 + 1 = 1
+     * Remaining length (max)   + 4 = 5
+     * Reason Code              + 1 = 6
```

Comment:
I didn't see the reason code mentioned in the fixed header. Are you just including it here because it is 'fixed' at 1 byte and required in the variable header?

I think it can technically be omitted if the remaining length is 0 BUT that really only saves one byte in this array. Not worth the calculation to determine if it is needed.